### PR TITLE
Charting refinements

### PR DESCRIPTION
**Pull request by hwbllmnn**
Includes various improvements, including:

* allow dynamic calculation of legend heights
* use more effects
* support initial hiding of lines
* support setting initial zoom
* bump dependency versions

### CODE REVIEW

File: src/LegendComponent/LegendComponent.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ class LegendComponent {
     if (!this.config.legendEntryMaxLength) {
       return 0;
     }
-    LabelUtil.handleLabelWrap(leg, ' g > text.legend-title', 25, 1.2, true, this.config.legendEntryMaxLength);
+    LabelUtil.handleLabelWrap(leg, ' g > text.legend-title', 25, 1.2, true, 20);
```
Comment (marcjansen) [inline src/LegendComponent/LegendComponent.js]: This looks odd. Don't we want this configurable?

File: src/LegendComponent/LegendComponent.js [hunk 2]
```diff
@@ -87,7 +87,7 @@ class LegendComponent {
...
     const count = leg.selectAll('tspan').size();
     return 14 * (count - 1);
```
Comment (marcjansen) [inline src/LegendComponent/LegendComponent.js]: 14?

File: src/ScaleUtil/ScaleUtil.js [hunk 3]
```diff
@@ -9,6 +9,8 @@ import scaleBand from 'd3-scale/src/band';
  */
 class ScaleUtil {
 
+  static EPSILON = 0.000001;
```
Comment (marcjansen) [inline src/ScaleUtil/ScaleUtil.js]: Isnt tjere a built in constant?

File: src/TimeseriesComponent/TimeseriesComponent.js [hunk 4]
```diff
@@ -684,7 +723,16 @@ class TimeseriesComponent {
    * Reset the zoom.
    */
   resetZoom() {
-    this.zoomBehaviour.transform(this.rootNode, identity);
+    this.rootNode.select('.timeseries-chart')
+      .transition().duration(750).call(this.zoomBehaviour.transform, identity);
```
Comment (marcjansen) [inline src/TimeseriesComponent/TimeseriesComponent.js]: Configurable?

File: src/TimeseriesComponent/TimeseriesComponent.js [hunk 5]
```diff
@@ -562,21 +588,25 @@ class TimeseriesComponent {
         const clip = g.append('g').attr('class', 'timeseries-clip');
         chartRoot = clip.append('g').attr('class', 'timeseries-chart');
       } else {
-        g.selectAll('.timeseries-data,.x-axis,.timeseries-line').remove();
+        g.selectAll('.timeseries-data,.x-axis,.x-grid-axis,.timeseries-line').remove();
```
Comment (marcjansen) [inline src/TimeseriesComponent/TimeseriesComponent.js]: Where do the `x-...` classes come from?

File: src/LegendComponent/LegendComponent.js (hunk 1 above)
Comment (hwbllmnn) [reply src/LegendComponent/LegendComponent.js]: We may want that to be configurable in the future, but the config option `legendEntryMaxLength` was actually misplaced here as the `distanceBuffer` parameter influences something else entirely.

File: src/LegendComponent/LegendComponent.js (hunk 2 above)
Comment (hwbllmnn) [reply src/LegendComponent/LegendComponent.js]: That's the default d3 text size (which obviously we might want to have configurable in the future).

File: src/ScaleUtil/ScaleUtil.js (hunk 3 above)
Comment (hwbllmnn) [reply src/ScaleUtil/ScaleUtil.js]: There is for example `Number.MIN_VALUE`, but using that with a log scale still results in lots of `NaN` values in the DOM, so I switched to this arbitrary low number.

File: src/TimeseriesComponent/TimeseriesComponent.js (hunk 4 above)
Comment (hwbllmnn) [reply src/TimeseriesComponent/TimeseriesComponent.js]: You mean the animation duration?

File: src/TimeseriesComponent/TimeseriesComponent.js (hunk 5 above)
Comment (hwbllmnn) [reply src/TimeseriesComponent/TimeseriesComponent.js]: That's the classes we set when rendering the axes/components.

**Comment by hwbllmnn** [comment]
Merging this now, we can make things configurable later with the current hardcoded values as defaults if we need to.

**Pull request merged**